### ISSUE TITLE
tC appears to lock up in wA

### ISSUE THREAD

**Issue by cckozie** [unfoldingWord/translationCore#4993]
1.0.0 (66fa79a)
[en_act.T.usfm.zip](https://github.com/unfoldingWord-dev/translationCore/files/2358174/en_act.T.usfm.zip)
Import the attached file 
Notice that the wA tool card indicates 99% complete
Load the project in wA 
Click on Word Alignment in the menu bar to return to the Tools page
Notice that the wA tool card still indicates 99% complete
Load the project in tW
Click on translationWords in the menu bar to return to the Tools page
Notice that the wA tool card now indicates significantly less than 99% complete (the % varies each time I run this test. Should this be a separate issue?)
Load the project in wA
The 'Changes have been detected...' alert is displayed and the app appears to be locked up for about 1 minute

**Comment by benjore** [comment]
Figure out what is happening here and then let me know what the problem is.

**Comment by da1nerd** [comment]
> FIXED with `usfm-js` 1.0.3

When I first opened the imported project the it says "Changes have been detected..."
It appears there is some inconsistency in handling usfm when importing projects, and stripping out usfm from the verses afterward.

See screenshot attached below.
![image](https://user-images.githubusercontent.com/166412/45188463-49e46000-b1e9-11e8-974c-4d3d762d629d.png)

When the project is first imported the alignment data is correctly extracted from the usfm. However, when the project is loaded some usfm artifacts left in the verse are causing wA to think the alignment data is missing token(s).

To summarize: the `removeUsfmMarkers` function used in wA is not correctly stripping out all the usfm markers. This method is using `usfm-js` and I'm not certain yet if this is due to improper parameters to the utility or a bug in `usfm-js` itself.

**Comment by da1nerd** [comment]
Another issue I discovered is during the import an extra source token is added to the alignment data.

e.g. in Acts 3:1 `τὸ` appears twice in the alignment data even though it only appears once in the imported verse.
This seems to indicates a problem in extracting alignment data from the USFM during import.

**Comment by da1nerd** [comment]
Discovering some more things here. When opening the project in tW the alignment data for a number of chapters is getting completely erased.

---
**Update 1**
Discovered this was due to a race condition while loading tool data.
Added check to determine if the tool had finished loading.

---
**Update 2**
This will be fixed in https://github.com/unfoldingWord-dev/translationCore/pull/5011 and https://github.com/translationCoreApps/wordAlignment/pull/75

**Comment by da1nerd** [comment]
@benjore do you want this to get into `1.0`? I should have some PRs ready for this today.

**Comment by da1nerd** [comment]
Two of the problems in this issue have been addressed.
The remaining issue is a bug in the project import that is producing [duplicate tokens](https://github.com/unfoldingWord-dev/translationCore/issues/4993#issuecomment-419270673) in the alignment data.

**Comment by da1nerd** [comment]
I have determined the extra tokens found in `3:1` and `12:20` are due to invalid alignments found in the imported usfm file. Therefore the import is working correctly, and wA is properly validating the alignments.

I believe we have had discussions about eventually bringing alignment validation to the project import so that users do not have to open the tool before they can see there are problems. However, that must wait until after https://github.com/unfoldingWord-dev/translationCore/issues/4672 has been completed.

**Comment by cckozie** [comment]
1.0.0 (dc79cf3)
I am no longer seeing the apparent lock up, but now the % complete on the wA tool card is showing 0%.

**Comment by PhotoNomad0** [comment]
@cckozie - found the problem.  Looks like I have a bug parsing the grc resources and building the index.json.  most bible's index.json have just chapter/verse counts, the grc OL has chapter/verse/word counts.  Looking into it

**Comment by cckozie** [comment]
Fixed in 1.0.0 (6966b10)
